### DISCUSSION TITLE
Pallas masking

### DISCUSSION THREAD

**Discussion by srush** [jax-ml/jax#17414] [Q&A]
I'm curious about masking in Pallas. 

Currently the examples have the masking done inside the kernel function. However in practice this requires a bunch of `program_id` algebra that I would rather not do in order to find boundaries. 

Is there any semantics to have the masking done in the blockspec? Ideally I could just say "fill oob with zeros" there and not have to ever use programid.

**Comment by sharadmv** [comment]
> Is there any semantics to have the masking done in the blockspec? Ideally I could just say "fill oob with zeros" there and not have to ever use programid.

It's an idea we have thrown around but haven't gotten around to implementing. It might be possible to be done completely automatically actually using a padding transformation.
